### ISSUE TITLE
Apollo Server socket connection issue when used with Apollo Federation (and make-fetch-happen)

### ISSUE THREAD

**Issue by pcoulso1** [apollographql/federation#2313]

**Comment by glasser** [comment]
Migrating this to federation as this is primarily about Gateway, though it is possible that one response might include improving documentation to encourage Apollo Server subgraph developers to increase their keepAliveTimeout. More thoughts to come later once I've checked with some folks who might have good ideas. Really great reproduction — thanks! One note is that IIUC make-fetch-happen's retry feature (which we do disable) does retry ECONNRESET by default... but not for POSTs.  I suspect that we should just retry ECONNRESET for non-mutations in Gateway. It's also quite possible that Apollo Router handles this better than Node/make-fetch-happen does!

**Comment by pcoulso1** [comment]
I think the root of this issue is with the make-fetch-happen library which has hardcoded freeSocketTimeout, therefore I've raised the following bug report at https://github.com/npm/make-fetch-happen/issues/199

**Comment by glasser** [comment]
Yeah, though I could understand if their response was "the proper way to configure HttpAgent parameters is to override the agent, not to copy all of HttpAgent's options into make-fetch-happen". Though... the mechanisms they give to override the agent only let you set one fixed agent rather than use their per-host agent logic, right?

**Comment by lfaryj** [comment]
Any news in this topic maybe? We are having the same issue, as everyone who uses gateway newer version, where u updated it to use make-fetch-happen instead of node-fetch I guess
